contracts/psm/PSM.sol

```diff
@@ -0,0 +1,293 @@
+    uint256 public sellFee; // sell fee rate
+    uint256 public buyFee; // buy fee rate
+    address public feeReceiver; // fee receiver address
+    address public lisUSD; // lisUSD address
+
+    uint256 public dailyLimit; // daily buy limit
+    uint256 public minSell; // min sell amount
+    uint256 public minBuy; // min buy amount
+
+    uint256 public lastBuyDay; // last buy day
+    uint256 public dayBuyUsed; // day buy used
+
+    uint256 public constant FEE_PRECISION = 10000;
+
+    bytes32 public constant MANAGER = keccak256("MANAGER"); // manager role
+    bytes32 public constant PAUSER = keccak256("PAUSER"); // pause role
+
+    event SetBuyFee(uint256 buyFee);
+    event SetSellFee(uint256 sellFee);
+    event SetFeeReceiver(address feeReceiver);
+    event BuyToken(address account, uint256 realAmount, uint256 fee);
+    event SellToken(address account, uint256 realAmount, uint256 fee);
+    event SetDailyLimit(uint256 dailyLimit);
+    event SetMinSell(uint256 minSell);
+    event SetMinBuy(uint256 minBuy);
+    event SetVaultManager(address vaultManager);
+    event EmergencyWithdraw(address token, uint256 amount);
+
+    /// @custom:oz-upgrades-unsafe-allow constructor
+    constructor() {
+        _disableInitializers();
+    }
+
+    /**
+      * @dev initialize contract
+      * @param _admin admin address
+      * @param _manager manager address
+      * @param _token token address
+      * @param _feeReceiver fee receiver address
+      * @param _lisUSD lisUSD address
+      * @param _sellFee sell fee
+      * @param _buyFee buy fee
+      * @param _minSell min sell amount
+      * @param _minBuy min buy amount
+      */
+    function initialize(
+        address _admin,
+        address _manager,
+        address _token,
+        address _feeReceiver,
+        address _lisUSD,
+        uint256 _sellFee,
+        uint256 _buyFee,
+        uint256 _dailyLimit,
+        uint256 _minSell,
+        uint256 _minBuy
+    ) public initializer {
+        require(_admin != address(0), "admin cannot be zero address");
+        require(_manager != address(0), "manager cannot be zero address");
+        require(_token != address(0), "token cannot be zero address");
+        require(_feeReceiver != address(0), "feeReceiver cannot be zero address");
+        require(_lisUSD != address(0), "lisUSD cannot be zero address");
+        require(_sellFee <= FEE_PRECISION, "sellFee must be less or equal than FEE_PRECISION");
+        require(_buyFee <= FEE_PRECISION, "buyFee must be less or equal than FEE_PRECISION");
+        __AccessControl_init();
+        __ReentrancyGuard_init();
+        __Pausable_init();
+        __UUPSUpgradeable_init();
+
+        _setupRole(DEFAULT_ADMIN_ROLE, _admin);
+        _setupRole(MANAGER, _manager);
+
+        token = _token;
+        sellFee = _sellFee;
+        buyFee = _buyFee;
+        feeReceiver = _feeReceiver;
+        lisUSD = _lisUSD;
+
+        dailyLimit = _dailyLimit;
```

Comment:
Do we need to check `dailyLimit` is higher than both `minSell` and `minBuy` on initialization?